src/ContinuosBondingERC20Token.sol

```diff
@@ -201,45 +208,54 @@ contract ContinuosBondingERC20Token is ERC20, ReentrancyGuard {
         currentEth = currentEth > address(this).balance ? address(this).balance : currentEth;
         isLpCreated = true;
 
-        //is this line necessary? can be removed
-        // _approve(address(this), address(router), currentTokenBalance);
-
-        address wNative;
-        uint256 liquidity;
-        if (poolType == LP_POOL.Uniswap) {
-            wNative = IUniswapV2Router02(router).WETH();
-
-            address pair = factory.getPair(address(this), wNative);
-            if (pair == address(0)) pair = factory.createPair(address(this), wNative);
-            IWETH(wNative).deposit{ value: currentEth }();
-            IWETH(wNative).transfer(pair, currentEth);
-            _transfer(address(this), pair, currentTokenBalance);
-            liquidity = IPair(pair).mint(address(this));
-        } else if (poolType == LP_POOL.TraderJoe) {
-            wNative = IJoeRouter02(router).WAVAX();
-
-            address pair = factory.getPair(address(this), wNative);
-            if (pair == address(0)) pair = factory.createPair(address(this), wNative);
-            IWETH(wNative).deposit{ value: currentEth }();
-            IWETH(wNative).transfer(pair, currentEth);
-            _transfer(address(this), pair, currentTokenBalance);
-            liquidity = IPair(pair).mint(address(this));
+        // Create the pool if it doesn't exist
+        (address token0, address token1) = address(this) < WETH ? (address(this), WETH) : (WETH, address(this));
+        address pool = nonfungiblePositionManager.createAndInitializePoolIfNecessary(token0, token1, 30000, _getSqrtPriceX96(currentEth, currentTokenBalance));
+
+        // Approve the position manager to spend tokens
+        _approve(address(this), address(nonfungiblePositionManager), currentTokenBalance);
+        IWETH(WETH).deposit{value: currentEth}();
+        IWETH(WETH).approve(address(nonfungiblePositionManager), currentEth);
+
+        // Add liquidity
+        INonfungiblePositionManager.MintParams memory params = INonfungiblePositionManager.MintParams({
+            token0: address(this),
+            token1: WETH,
+            fee: 3000,
+            tickLower: TickMath.MIN_TICK,
+            tickUpper: TickMath.MAX_TICK,
+            amount0Desired: currentTokenBalance,
+            amount1Desired: currentEth,
+            amount0Min: 0,
+            amount1Min: 0,
+            recipient: address(this),
+            deadline: block.timestamp
+        });
+
+        (uint256 tokenId, uint128 liquidity, uint256 amount0, uint256 amount1) = nonfungiblePositionManager.mint(params);
+
+        // Transfer the NFT to the treasury if feeRecipient is not zero
+        if (IBondingERC20TokenFactory(factory).feeRecipient() != address(0)) {
+            nonfungiblePositionManager.transferFrom(address(this), TREASURY_ADDRESS, tokenId);
```

Comment:
why not just directly mint the position nft to `IBondingERC20TokenFactory(factory).feeRecipient()`? 

currently, we are minting to current contract and transferring to `IBondingERC20TokenFactory(factory).feeRecipient()`. Can't we reduce it to single step by changing `recipient`?